### PR TITLE
[trainer] feat: make lora parameters dtype configurable

### PR DESCRIPTION
**Pull request by zhtmike**
### What does this PR do?
- Convert the LoRA parameters to FP32 even when `model_dtype=bfloat16`. The optimizer buffer should always be in FP32 to avoid precision loss caused by `bfloat16` buffers under the PyTorch native optimizer.
- This should result in minimal memory increase since LoRA weights occupy only a very small portion of the entire model.

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `vllm_omni`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `diffusion`, `omni`, `tests`, `docker`
  - If this PR involves multiple modules, separate them with `,` like `[diffusion, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][diffusion, fsdp] feat: new rollout scheduler`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/verl-project/verl-omni/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update the documentation.
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl-omni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: verl_omni/workers/engine/fsdp/diffusers_impl.py [hunk 1]
```diff
@@ -250,6 +252,17 @@ def _build_lora_module(self, module):
             }
             module.add_adapter(LoraConfig(**lora_config))
 
+        # Convert LoRA parameters to fp32 for numerical stability during training.
+        for name, param in module.named_parameters():
+            if param.requires_grad:
+                orig_dtype = param.dtype
+                param.data = param.data.to(torch.float32)
```
Comment (SamitHuang) [inline verl_omni/workers/engine/fsdp/diffusers_impl.py]: Let's make model weight dtype configurable

Comment (zhtmike) [reply verl_omni/workers/engine/fsdp/diffusers_impl.py]: ok, let me validate it first

**Review by SamitHuang — APPROVED** [review]
LGTM overall. Can update the docs or provide a example script for how to set fp32 training

**Comment by SamitHuang** [comment]
if there are wandb run curves, better attach to PR for reference

**Comment by SamitHuang** [comment]
Should we add a cpu unit test to cover it?

**Comment by zhtmike** [comment]
> if there are wandb run curves, better attach to PR for reference

no wandb log currently sorry, but I am using this feature for bagel training

**Comment by zhtmike** [comment]
> LGTM overall. Can update the docs or provide a example script for how to set fp32 training

done. updated doc.

**Comment by zhtmike** [comment]
> Should we add a cpu unit test to cover it?

I think the `peft` should already takes in charge for this feature

File: docs/algo/flowgrpo.md
Comment (SamitHuang) [inline docs/algo/flowgrpo.md:1]: is it compatible for mixgrpo and grpo-guard?

Comment (zhtmike) [reply docs/algo/flowgrpo.md:1]: yes, might extract them into a seperate section @AndyZhou952
